packages/server-wallet/e2e-test/e2e.test.ts

```diff
@@ -65,8 +79,40 @@ describe('e2e', () => {
     expect(channel.status).toBe('opening');
     expect(channel.turnNum).toBe(0);
   });
+});
 
-  it('can update pre-existing channel, send signed state via http', async () => {
+describe('pinging', () => {
+  let channelId: string;
+
+  const triggerPings = async (numPings?: number): Promise<void> => {
+    let args = [
+      'ts-node',
+      'e2e-test/e2e-utils/ping.ts',
+      '--database',
+      'ping',
+      '--channels',
+      channelId,
+    ];
+
+    if (numPings) args = args.concat(['--numPings', numPings.toString()]);
+
+    const pingScript = childProcess.spawn(`yarn`, args);
+    pingScript.on('error', logger.error);
+
+    await new Promise(resolve => {
+      pingScript.on('exit', _data => resolve());
+
+      // FIXME: I would expect `.on('exit', resolve)` to work, but it does not,
+      // jest warns 'Jest did not exit one second after the test run has completed.'
+      // This indicates to me that the spawned process does not exit, even after logging 'DONE'
+      pingScript.stdout.on('data', data => data.toString() === 'DONE\n' && resolve());
```

Comment:
:( If we have to use `DONE` as a magic keyword could we make it a constant? I'm hoping we can figure out a better way to handle this.

Reply:
Absolutely. The best thing is to sort out what's going on, though.

Reply:
I think this will resolve it. (It does locally)

https://github.com/statechannels/statechannels/pull/2417/commits/36041ad2a62198a17db2691555f93b6d4a06626a